presto-docs/src/main/sphinx/gcp/storage.rst

```diff
@@ -0,0 +1,216 @@
+However, in most cases data is not publicly available, and the Presto cluster needs to have access to it.
+This is typically achieved by creating a service account which has permissions to access your data.
+You can do this on the `service accounts page in GCP <https://console.cloud.google.com/projectselector2/iam-admin/serviceaccounts>`_.
+Once you create a service account, create a key for it and download the key in JSON format.
+
+Configure Cloud Storage connector
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Access to Cloud Storage data is possible thanks to `Hadoop Cloud Storage connector <https://cloud.google.com/dataproc/docs/concepts/connectors/cloud-storage>`_.
+It allows Presto (and other components using Hadoop) to access files in GCS via the ``gs://`` protocol.
+
+There are two ways to configure this connector:
+
+* Simplified Presto-specific configuration via Hive catalog properties (``hive.properties``)
+* Full configuration for the entire Hadoop ecosystem
+
+The simplified configuration method is useful if only Presto needs to access files in GCS, and is described in the `Hive Connector`_ section.
+If another Hadoop-based component (e.g. Hive metastore) also needs to access these files, it will be necessary to perform full configuration.
+
+.. _Full GCS configuration:
+
+**Full GCS configuration:**
+
+Google Cloud Storage access is configured in ``core-site.xml`` and will be used by all components using Apache Hadoop.
+Here's an example set of properties responsible for GCS access:
+
+.. code-block:: XML
+
+    <property>
+        <name>fs.gs.impl</name>
+        <value>com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem</value>
+        <description>The FileSystem for gs: (GCS) uris.</description>
+    </property>
+    <property>
+        <name>fs.AbstractFileSystem.gs.impl</name>
+        <value>com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS</value>
+        <description>
+            The AbstractFileSystem for gs: (GCS) uris. Only necessary for use with Hadoop 2.
+        </description>
+    </property>
+    <property>
+        <name>fs.gs.project.id</name>
+        <value></value>
+        <description>
+            Required. Google Cloud Project ID with access to configured GCS buckets.
+        </description>
+    </property>
+    <!--
+    The following property is required when Presto is NOT running in a GCE VM.
+    The connector will look to see if it is running on a GCE VM with some level of GCS access in it's service account scope, and use that service account.
+    -->
+    <property>
+        <name>fs.gs.auth.service.account.json.keyfile</name>
+        <value></value>
+        <description>
+            The path to the json keyfile for the service account.
+        </description>
+    </property>
+    <!--
+    The following property is required only when Presto must run as different user than Hive metastore and Hive metastore uses StorageBasedAuthorization.
```

Comment:
Is this configuration needed by Presto or needed by the Hive metastore? If the former, we should add a config property for it (or probably just set it correctly by default). If the latter, we should mention it here.

Reply:
It's needed by Hive Metastore when it's doing a POSIX permissions check. I will make this clear.